### PR TITLE
Clean up System.Private.Xml tests

### PR DESCRIPTION
**Pull request by sepidehMS**
Resolves https://github.com/dotnet/corefx/issues/14071
Does the following:
1. Remove commented tests that no longer make sense (either due to APIs not being part of netstandard2.0, or the action not being allowed anymore due to security)
2. Fix and re-enable good tests
3. Re-add some test properties that were not previously ported (particularly in XSLT tests)
4. Remove dead code
5. Other general code cleanup

cc: @danmosemsft @AlexGhiondea

### CODE REVIEW

File: src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs [hunk 1]
```diff
@@ -125,7 +125,7 @@ internal XmlResolver GetXmlResolver()
         //notice we must keep GetXmlResolver() to avoid dead lock when init System.Config.ConfigurationManager
         internal XmlResolver GetXmlResolver_CheckConfig()
         {
-            if (!IsXmlResolverSet)
+            if (!LocalAppContextSwitches.DontProhibitDefaultResolver && !IsXmlResolverSet)
```
Comment (sepidehMS) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: We need to check for the value of this switch to give the option to use `<xsd:import schemaLocation"..." />` to import external xsd inside another one. This is not allowed by default.

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

File: src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs (hunk 1 above)
Comment (danmosemsft) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: What do you think of reversing the sense, so it's just 'ProhibitDefaultResolver'

Negative names consistently cause confusion, and worse when they are getting used in a negated context like this! 

---
In reply to: [94083128](https://github.com/dotnet/corefx/pull/14751#discussion_r94083128) [](ancestors = 94083128)

**Comment by danmosemsft** [comment]
>                 //XmlExceptionVerifier.IsExceptionOk(e, "Sch_InvalidWhitespaceInEmpty");

this one?

---
Refers to: src/System.Private.Xml/tests/XmlSchema/XmlSchemaValidatorApi/ValidateWhitespace_String.cs:104 in 3d6c09a. [](commit_id = 3d6c09a9ac0b7f32964820334f00edc7fbec1e91, deletion_comment = False)

**Comment by danmosemsft** [comment]
>                     //XmlExceptionVerifier.IsExceptionOk(e, "Sch_UnresolvedKeyref", new string[] { "3", "numberKey" });

this one?

---
Refers to: src/System.Private.Xml/tests/XmlSchema/XmlSchemaValidatorApi/ValidateText_EndElement.cs:637 in 3d6c09a. [](commit_id = 3d6c09a9ac0b7f32964820334f00edc7fbec1e91, deletion_comment = False)

**Comment by danmosemsft** [comment]
>             foreach (bool enableDebug in new bool[] { /*true,*/ false }) //ActiveIssue(10023)

there are 2 instances of this still. I guess this should be // XSLT debugging not supported in Core

---
Refers to: src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/Errata4.cs:220 in 3d6c09a. [](commit_id = 3d6c09a9ac0b7f32964820334f00edc7fbec1e91, deletion_comment = False)

File: src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs (hunk 1 above)
Comment (sepidehMS) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: @danmosemsft I believe there was an unwritten rule with `AppContextSwitches` that the default value needs to be false and that's why we initially named the switch this way. @AlexGhiondea probably has more context on this though.

Comment (danmosemsft) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: Well "AllowDefaultResolver" then?

**Comment by danmosemsft** [comment]
Good work @sepidehMS thank you for doing all this tedious cleanup. LGTM.

If you know how, it would be great to run these locally on desktop also. Otherwise likely some failures may have crept in. If you do that I would love to know how to do it myself also.

**Comment by sepidehMS** [comment]
@danmosemsft Thanks for reviewing this. Made requested changes. Will try to figure out how to run desktop tests locally.

I also had to disable two tests in `System.Xml.XmlSchemaSet.Tests.csproj` as I'm seeing inconsistent behavior when setting `TestSwitch.LocalAppContext.DisableCaching` to `true`. This property does not get the right value in some of the runs. Going to re-open https://github.com/dotnet/corefx/issues/14064.

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Pull request merged**